core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -0,0 +1,220 @@
+
+            git.add().addFilepattern("pom.xml").call();
+            git.commit().setMessage("Imports pom.xml file").setSign(false).setCommitter(committer).call();
+
+            final HasUnreleasedProductionChangesProbe probe = getSpy();
+            final ProbeResult result = probe.apply(plugin, ctx);
+
+            assertThat(probe.apply(plugin, ctx))
+                .usingRecursiveComparison()
+                .comparingOnlyFields("id", "status")
+                .isEqualTo(result.success(HasUnreleasedProductionChangesProbe.KEY, ""));
+        }
+    }
+
+    @Test
+    void commitOnReadmeFileAfterReleaseDateShouldReturnSuccess() throws IOException, GitAPIException {
+        final Path repository = Files.createTempDirectory("test-foo-bar");
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final String scmLink = "https://test-server/jenkinsci/test-repo/test-folder";
+
+        ZonedDateTime releaseTimestamp = ZonedDateTime.now().minusHours(38);
+        when(plugin.getReleaseTimestamp()).thenReturn(releaseTimestamp);
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        when(ctx.getScmRepository()).thenReturn(repository);
+        when(plugin.getScm()).thenReturn(scmLink);
+
+        try (Git git = Git.init().setDirectory(repository.toFile()).call()) {
+
+            final Path pom = Files.createFile(repository.resolve("README.md"));
+
+            // creating commit
+
+            PersonIdent defaultCommitter = new PersonIdent(git.getRepository());
+            PersonIdent committer = new PersonIdent(defaultCommitter, Date.from(plugin.getReleaseTimestamp().plusDays(1).toInstant()));
+
+            git.add().addFilepattern("README.md").call();
+            git.commit().setMessage("Updated README.md file").setSign(false).setCommitter(committer).call();
+
+            final HasUnreleasedProductionChangesProbe probe = getSpy();
+            final ProbeResult result = probe.apply(plugin, ctx);
+
+            assertThat(probe.apply(plugin, ctx))
+                .usingRecursiveComparison()
+                .comparingOnlyFields("id", "status")
+                .isEqualTo(result.success(HasUnreleasedProductionChangesProbe.KEY, ""));
+        }
+    }
+
+    @Test
+    void checkThatCommitOnSrcPathBeforeReleaseDateReturnsSuccess() throws IOException, GitAPIException {
+        final Path repository = Files.createTempDirectory("test-foo-bar");
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final String scmLink = "https://test-server/jenkinsci/test-repo/test-folder";
+
+        ZonedDateTime releaseTimestamp = ZonedDateTime.now().minusHours(38);
```

Comment:
Do we need `.minusHours(38)`?
Especially when you're already performing a git commit on `plugin.getReleaseTimestamp().minusDays(1)` which will always be 1 day behind the release date. So maybe `.minusHours(38)` is redundant.

Reply:
In the last meeting, Adrien suggested adding `.minusHours(...)` to deal with the time difference as far as I remember. That is why I have added them. 

I will remove it if you deem it redundant.

Reply:
This is incorrect no? If the release happened 38hours ago, and you add commits on files now, your commits are _after_ the release date, correct? This is not what the test name implied. 
```suggestion
        ZonedDateTime releaseTimestamp = ZonedDateTime.now().plusHours(38);
```

Reply:
I understood what my mistake was.

My test case shouldn't have passed in the old case.